### PR TITLE
fix(users): Allow accepting invites for `org_admin`s

### PR DESCRIPTION
**Pull request by ThisIsMani**
## Type of Change

- [x] Bugfix
- [ ] New feature
- [ ] Enhancement
- [ ] Refactoring
- [ ] Dependency updates
- [ ] Documentation
- [ ] CI/CD

## Description

Currently accepting invites for org_admin will not work. This PR fixes it.

### Additional Changes

- [ ] This PR modifies the API contract
- [ ] This PR modifies the database schema
- [ ] This PR modifies application configuration/environment variables

## Motivation and Context

Closes [#6254](https://github.com/juspay/hyperswitch/issues/6254)

## How did you test it?

```
curl 'http://localhost:8080/user/accept_invite_from_email' \
  -H 'authorization: Bearer SPT with accept_invite_from_email' \
  --data-raw '{"token":"email token"}'
```
Response:
```json
{
    "token": "JWT",
    "token_type": "user_info"
}
```

## Checklist

- [x] I formatted the code `cargo +nightly fmt --all`
- [x] I addressed lints thrown by `cargo clippy`
- [x] I reviewed the submitted code
- [ ] I added unit tests for my changes where possible

### CODE REVIEW

File: crates/router/src/utils/user_role.rs [hunk 1]
```diff
@@ -255,12 +255,54 @@ pub async fn get_lineage_for_user_id_and_entity_for_accepting_invite(
...
+                    version: None,
+                    status: Some(UserStatus::InvitationSent),
+                    limit: None,
+                })
+                .await
+                .change_context(UserErrors::InternalServerError)?
+                .into_iter()
+                .collect::<HashSet<_>>();
+
+            if user_roles.len() > 1 {
```
Comment (apoorvdixit88) [inline crates/router/src/utils/user_role.rs:284]: We can evaluate this?

**Review by apoorvdixit88 — DISMISSED** [review]

File: crates/router/src/utils/user_role.rs [hunk 2]
```diff
@@ -255,12 +255,54 @@ pub async fn get_lineage_for_user_id_and_entity_for_accepting_invite(
...
-        id_type::MerchantId,
+        Option<id_type::MerchantId>,
         Option<id_type::ProfileId>,
     )>,
 > {
     match entity_type {
-        EntityType::Organization => Err(UserErrors::InvalidRoleOperation.into()),
+        EntityType::Organization => {
+            let org_id = id_type::OrganizationId::try_from(std::borrow::Cow::from(entity_id))
+                .change_context(UserErrors::InternalServerError)?;
```
Comment (SanchithHegde) [inline crates/router/src/utils/user_role.rs:265]: Can we have a context message here (with `attach_printable` / `attach_printable_lazy`) and other places where we're raising internal server errors?

File: crates/router/src/utils/user_role.rs (hunk 1 above)
Comment (ThisIsMani) [reply crates/router/src/utils/user_role.rs:284]: This is correct as for a particular entity_id, there can only be 1 invite and not more than that.

File: crates/router/src/utils/user_role.rs (hunk 2 above)
Comment (ThisIsMani) [reply crates/router/src/utils/user_role.rs]: - I've gone through the code and checked wherever we are throwing `InternalServerError`. All of the places are mostly
  1. Data problems, which we can find using DB efficiently as we don't have the whole data of the object.
  2. Logical errors or Unreachable errors, which can be found out if we see the code at the line number at which the error is logged.
- In my opinion, adding `attach_printable`s without any important debug information will pollute the code. So, that's why I don't prefer adding them unless they are adding any value to the log.
- In the place where you commented, there was no need to throw 500 and I changed that code.

**Review by SanchithHegde — APPROVED** [review]

**Review by racnan — APPROVED** [review]
lets remove `Entity` type from email and use one `Entity` across repo.

**Review by apoorvdixit88 — APPROVED** [review]

**Review by Riddhiagrawal001 — APPROVED** [review]

**Pull request merged**